src/note.rs

```diff
@@ -19,12 +21,44 @@ pub use self::commitment::{ExtractedNoteCommitment, NoteCommitment};
 pub(crate) mod nullifier;
 pub use self::nullifier::Nullifier;
 
+// We know that `pallas::Base` doesn't allocate internally.
+memuse::impl_no_dynamic_usage!(Rho);
+
+/// The randomness used to construct a note.
+///
+/// The [`Rho`] value for a note should always be constructed from the revealed nullifier of the
+/// paired spend in the process of creating an [`Action`].
+///
+/// [`Action`]: crate::action::Action
+#[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub struct Rho(pub(crate) pallas::Base);
+
+impl Rho {
+    /// Constructs the [`Rho`] value to be used to construct a new note from the revealed nullifier
+    /// of the note being spent in the [`Action`] under construction.
+    ///
+    /// [`Action`]: crate::action::Action
+    pub fn from_paired_spend_revealed_nf(nf: Nullifier) -> Self {
```

Comment:
We can't use `from_action` because the note that will be created using this `Rho` is used to create the action.

Reply:
We create that note inside the `orchard` crate, where we can use crate-private functions. To downstream users of the `orchard` crate, I think we do not need to expose that functionality: actions should be created as units, rather than piecemeal.

Reply:
I see, okay, this function can be crate-private and then I'll add the action-based constructor instead.